src/io/config.cpp

```diff
@@ -110,28 +110,30 @@ void OverallConfig::GetTaskType(const std::unordered_map<std::string, std::strin
 }
 
 void OverallConfig::CheckParamConflict() {
+  GBDTConfig* gbdt_config = dynamic_cast<GBDTConfig*>(boosting_config);
   if (network_config.num_machines > 1) {
     is_parallel = true;
   } else {
     is_parallel = false;
-    dynamic_cast<GBDTConfig*>(boosting_config)->tree_learner_type =
-                                                TreeLearnerType::kSerialTreeLearner;
+    gbdt_config->tree_learner_type = TreeLearnerType::kSerialTreeLearner;
   }
 
-  if (dynamic_cast<GBDTConfig*>(boosting_config)->tree_learner_type ==
-                                                TreeLearnerType::kSerialTreeLearner) {
+  if (gbdt_config->tree_learner_type == TreeLearnerType::kSerialTreeLearner) {
     is_parallel = false;
     network_config.num_machines = 1;
   }
 
-  if (dynamic_cast<GBDTConfig*>(boosting_config)->tree_learner_type ==
-                                                 TreeLearnerType::kSerialTreeLearner ||
-    dynamic_cast<GBDTConfig*>(boosting_config)->tree_learner_type ==
-                                                 TreeLearnerType::kFeatureParallelTreelearner) {
+  if (gbdt_config->tree_learner_type == TreeLearnerType::kSerialTreeLearner ||
+    gbdt_config->tree_learner_type == TreeLearnerType::kFeatureParallelTreelearner) {
     is_parallel_find_bin = false;
-  } else if (dynamic_cast<GBDTConfig*>(boosting_config)->tree_learner_type ==
-                                                 TreeLearnerType::kDataParallelTreeLearner) {
+  } else if (gbdt_config->tree_learner_type == TreeLearnerType::kDataParallelTreeLearner) {
     is_parallel_find_bin = true;
+    if (gbdt_config->tree_config.histogram_pool_size >= 0) {
+      Log::Error("Auto set histogram_pool_size to non-limit, to reduce communication cost");
```

Comment:
" Histogram LRU queue was enabled for reducing communication cost. ( histogram_pool_size = %f )", tree_config.histogram_pool_size